csrc/scheduler/vectorize_helper.cpp

```diff
@@ -239,9 +246,16 @@ ContiguousInnerDimensionsMapper::ContiguousInnerDimensionsMapper(
       if (std::find(reference_ids.begin(), reference_ids.end(), id) !=
           reference_ids.end()) {
         reordered_root.push_back(id);
-        // Initiailze the extent for the mapped iter domain
+        // Initialize the extent for the mapped iter domain
         ProjectedExtent pe;
-        pe.multiplyNumeratorValue(commonOrConstExtent(ca_map_, id));
+        auto ext = commonOrConstExtent(ca_map_, id);
+        if (ext->isConstInt() && ext->evaluateInt() == 0) {
+          // A size-zero extent ID will be predicated out always, so it should
+          // not affect the projected extent calculation
+          continue;
```

Comment:
@zasdfgbnm I know you've been looking at this code lately also, in relation to #393 . I am trying to avoid bad behavior when an empty TensorView with size zero extent is encountered. I currently am just pretending it doesn't exist here for the purposes of computing the projected extent. But I'm a little uncertain of whether that is enough, or if it's even safe. I think I have an idea of how it's supposed to work but I'd appreciate you having a look if you have some time.

Reply:
I think we should clean up this entire thing with expr simplifier, and I am thinking about doing so after #393.

Reply:
I think for the case of zero extent, as long as this does not throw an error, whatever value it returns does not matter, because we will not do any memory read/write anyway. Am I correct?

Reply:
If a tensor has zero elements, should we just stop propagation and assume that tensor is OK with arbitrary vectorization size?